### PR TITLE
schemas: remove fields on documents, deposits and users

### LINKED ISSUE #752: In a shared repository, fields `subdivision` and `collection` should not be available
It applies to documents, deposits and users. In the user editor, there is a subdivision facet, which should also be deactivated.

See also #751

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Moves the schema processing to dedicated classes.
* Adds fixtures for organisations and users.
* Removes collections, subdivisions, customField1, customField2
  and customField3 on documents and deposits schema.
* Removes subdivison on users for shared organisation.
* Closes https://github.com/rero/sonar/issues/752.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: sonar/json_schemas/deposits_json_schema.py [hunk 1]
```diff
@@ -32,14 +32,33 @@ def process(self):
...
+        if current_user_record:
+            organisation = current_user_record.replace_refs() \
+                .get('organisation')
+
+        if not current_user_record or (current_user_record.is_moderator and \
+            organisation.get('isDedicated')):
             return schema
 
+        # Remove some fields on json for the shared organisation
+        if organisation.get('isShared'):
```
Comment (jma) [inline sonar/json_schemas/deposits_json_schema.py:45]: Is `organisation` a list or a simple value? L. 35 suggest a list but this line suggest a simple value.

Comment (jma) [reply sonar/json_schemas/deposits_json_schema.py:45]: To be safer, we can keep the fields only for dedicated:
```
if not organisation.get('isDedicated', False):
```

File: sonar/json_schemas/documents_json_schema.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        :returns: The processed schema.
+        """
+        schema = super().process()
+
+        if not current_user_record:
+            return schema
+
+        # Get Organisation for the current logged user
+        organisation = current_user_record.replace_refs().get('organisation')
```
Comment (jma) [inline sonar/json_schemas/documents_json_schema.py:39]: You can put de default value for the `get` function.

File: sonar/json_schemas/projects_json_schema.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    def process(self):
+        """Projects JSON schema custom process.
+
+        :returns: The processed schema.
+        """
+        schema = super().process()
+
+        # Remove modes fields if user does not have superuser role.
+        if (current_user and not current_user.is_anonymous) \
```
Comment (jma) [inline sonar/json_schemas/projects_json_schema.py:38]: This condition can be removed as `current_user_record` should be None in this case.

File: tests/ui/test_views.py [hunk 4]
```diff
@@ -134,52 +134,74 @@ def test_logged_user(app, client, superuser, admin, moderator, submitter,
     assert not res.json['metadata']['permissions']['deposits']['list']
 
 
-def test_schemas(client, admin, user, submitter, moderator):
-    """Test JSON schemas endpoint."""
-    res = client.get(url_for('sonar.schemas', record_type='documents'))
-    assert res.status_code == 200
-    assert res.json['schema']['properties'].get('organisation')
+def test_schema_no_existing(client, user):
```
Comment (jma) [inline tests/ui/test_views.py:137]: Can be replaced by `def test_non_existing_schema`, you can also change the doc string.

**Review by jma — APPROVED** [review]

File: sonar/json_schemas/deposits_json_schema.py (hunk 1 above)
Comment (Garfield-fr) [reply sonar/json_schemas/deposits_json_schema.py:45]: Error in my code: `organisation = {}`

File: sonar/json_schemas/documents_json_schema.py (hunk 2 above)
Comment (Garfield-fr) [reply sonar/json_schemas/documents_json_schema.py:39]: I don't have a default value? What do you want me to put?

**Review by mmo — APPROVED** [review]

**Comment by mmo** [comment]
@pronguen : 

> [ ] Logged in as an admin of a shared organisation
> * [ ]  I am still able to add the local field to a document from the editor. I shouldn't.

I could not reproduce this behaviour. My test is successful.

> * [ ]  In the user editor, I still have the subdivision field. I shouldn't.

I confirm this behaviour, but it's true that the **user form** was not mentioned in the specification. It has now been added.

**Review by mmo — CHANGES_REQUESTED** [review]
See my comments above: https://github.com/rero/sonar/pull/758#issuecomment-1038819175

**Review by mmo — APPROVED** [review]

**Pull request merged**